packages/insomnia-components/components/sidebar.js

```diff
@@ -140,204 +213,637 @@ const StyledPanel: React.ComponentType<{}> = styled(motion.div)`
 `;
 
 function Sidebar(props: Props) {
-  // Temp garbage for easing/transition/sequencing dial-in
-  const [visible, setVisible] = useState(false);
-  const [visible2, setVisible2] = useState(false);
-  const [visible3, setVisible3] = useState(true);
-  const [visible4, setVisible4] = useState(true);
-  const toggleVisible = () => setVisible(!visible);
-  const toggleVisible2 = () => setVisible2(!visible2);
-  const toggleVisible3 = () => setVisible3(!visible3);
-  const toggleVisible4 = () => setVisible4(!visible4);
+  // Section Expansion & Filtering
+  // Info
+  const [infoSec, setInfoSec] = useState(false);
+  const toggleInfoSec = () => setInfoSec(!infoSec);
```

Comment:
Quick note: these callbacks will always create a new instance every time, causing the child components to update unnecessarily due to a prop change, even though most of the `toggle` callbacks will remain unchanged.

I suggest the `useCallback` [hook ](https://reactjs.org/docs/hooks-reference.html#usecallback) to set these, which returns a memoized callback if the dependencies (in this case, `infoSec` and `setInfoSec`) don't change. It's a little more verbose, but you can write your own hook to wrap that logic, eg:

```js
const useToggle = (state, set) => React.useCallback(() => set(!state), [set, state]);
```

Resulting in 
```suggestion
  const toggleInfoSec = useToggle(infoSec, setInfoSec);
```